.github/workflows/publish.yml

```diff
@@ -0,0 +1,56 @@
+name: Publish
+on:
+  release:
+    types:
+      - published
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: v4
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+      - uses: actions/cache@v1
+        id: yarn-cache
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+      - name: Use Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: '10.x'
+      - name: Install dependencies
+        run: yarn install
+      - name: Build
+        run: yarn build
+      - name: Lint
+        run: yarn lint
+      - name: Test
+        run: yarn test
+      - name: Get the version
+        id: vars
+        run: echo ::set-output name=tag::$(echo ${GITHUB_REF#"refs/tags/"})
+      - name: Bump version
+        run: npm version ${{ steps.vars.outputs.tag }} --no-git-tag-version
+      - name: Commit files
+        run: |
+          git config --local user.email "action@github.com"
+          git config --local user.name "GitHub Action"
+          git commit -m "Bump to ${{ steps.vars.outputs.tag }}" -a
+          git tag -f ${{ steps.vars.outputs.tag }}
+      - name: Push changes
+        uses: ad-m/github-push-action@master
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          branch: v4
+          force: true
```

Comment:
why need force?

Reply:
The git tag will be added by publishing the release notes. However, we will do bump the version of `package.json` and push a git commit for it in this publish workflow. Therefore, the git tag will be behind this commit. We need to move the git tag on this commit via force push.

Reply:
is it possible to let github not to add a tag when it add release note?

Reply:
Nope, that will get the error "We weren't able to create the release for you. Make sure you have a valid tag."